container-core/src/main/java/com/yahoo/container/jdisc/state/StateHandler.java

```diff
@@ -212,6 +216,56 @@ private ObjectNode buildJsonForConsumer(String consumer) {
         return ret;
     }
 
+    private byte[] buildPrometheusForConsumer(String consumer) {
+        var snapshot = getSnapshot();
+        if (snapshot == null)
+            return new byte[0];
+
+        var timestamp = snapshot.getToTime(TimeUnit.MILLISECONDS);
+        var builder = new StringBuilder();
+        builder.append("# NOTE: THIS API IS NOT INTENDED FOR PUBLIC USE\n");
+        for (var tuple : collapseMetrics(snapshot, consumer)) {
+            var dims = toPrometheusDimensions(tuple.dim);
+            var metricName = prometheusSanitizedName(tuple.key) + "_";
+            if (tuple.val instanceof GaugeMetric gauge) {
+                appendPrometheusEntry(builder, metricName + "max", dims, gauge.getMax(), timestamp);
+                appendPrometheusEntry(builder, metricName + "sum", dims, gauge.getSum(), timestamp);
+                appendPrometheusEntry(builder, metricName + "count", dims, gauge.getCount(), timestamp);
+                if (gauge.getPercentiles().isPresent()) {
+                    for (Tuple2<String, Double> prefixAndValue : gauge.getPercentiles().get()) {
+                        appendPrometheusEntry(builder, metricName + prefixAndValue.first + "percentile", dims, prefixAndValue.second, timestamp);
+                    }
+                }
+            } else if (tuple.val instanceof CountMetric count) {
+                appendPrometheusEntry(builder, metricName + "count", dims, count.getCount(), timestamp);
+            }
+        }
+        return builder.toString().getBytes(UTF_8);
+    }
+
+    private void appendPrometheusEntry(StringBuilder builder, String metricName, String dimension, Number value, long timeStamp) {
+        builder.append("# HELP ")
+                .append(metricName)
+                .append("\n# TYPE ")
+                .append(metricName)
+                .append(" untyped\n");
+
+        builder.append(metricName)
+                .append("{").append(dimension).append("}")
+                .append(" ").append(sanitizeIfDouble(value)).append(" ")
+                .append(timeStamp).append("\n");
+    }
+
+    private String toPrometheusDimensions(MetricDimensions dimensions) {
+        if (dimensions == null) return "";
+        StringBuilder builder = new StringBuilder();
+        dimensions.forEach(entry -> {
+            var sanitized = prometheusSanitizedName(entry.getKey()) + "=\"" + entry.getValue() + "\",";
```

Comment:
Unless `entry.getValue()` is pre-sanitized we might want to add some escaping here to ensure our label values are always edible:
> `label_value` can be any sequence of UTF-8 characters, but the backslash (`\`), double-quote (`"`), and line feed (`\n`) characters have to be escaped as `\\`, `\"`, and `\n`, respectively